execution-env.md

```diff
@@ -56,31 +56,56 @@ An access is considered explicitly granted only if:
 *   It is a read access to a resource attached for reading to the ExecutionContext
 *   It is a write access to a resource attached for writing to the ExecutionContext
 
+### Out-of-bounds Access Behavior
+
 Typically a WebGPU implementation can make sure that the _App_ can only access resources provided in the ExecutionContext by validating the code in the shader module.
 However this only guarantees that the correct resources are accessed, not that they are accessed in bounds.
-This is an issue because resource accesses are often lowered to pointer arithmetic so out of bounds accesses could result in accesses outside the ExecutionContext.
-For this reason out of bound accesses to resources must be prevented.
 
-Validating accesses are in bounds at shader module or pipeline creation time isn't possible because it is equivalent to the halting problem.
-Instead a WebGPU implementation must ensure out of bounds accesses don't cause accesses outside of the ExecutionContext.
-This will typically be done by instrumenting the generated code, and because of the performance sensitive nature of it, implementation will have a choice in the behavior of each out of bounds access.
-Similarly to OpenGL's `GL_KHR_robust_buffer` extension, out of bounds accesses must produce one of the following behaviors:
+A WebGPU implementation must ensure out of bounds accesses don't cause accesses outside of the ExecutionContext.
+This will typically be done by instrumenting the generated code. Because this code is highly performance sensitive, the WebGPU implementation is granted multiple options for enforcing security.
+
+The following rules are based on those defined by the
+*robustBufferAccess* feature in the [Vulkan 1.0
+specification](https://www.khronos.org/registry/vulkan/specs/1.0/html/vkspec.html#features-features).
+
+Out-of-bounds buffer loads _may_ result in any of the following behaviors:
 
-* Be discarded for writes
-* Access any location within the resource for reads and writes
-* Return zero values for reads or (0, 0, 0, X) with X being 0, 1, -1, or extrema for integers, or -0.0, +0.0, -1.0, +1.0 for floating point values
-* Atomics can return undefined values.
+* Return values from anywhere within the memory range(s) bound to the
+  buffer (possibly including bytes of memory past the end of the
+  buffer, up to the end of the bound range).
 
-The robust resource access behavior of OpenGL is extended to allow for trapping behavior as well:
+* Return zero values, or _(0,0,0,x)_ vectors for vector reads,
+  where x is a valid value represented in the type of the vector
+  components and _may_ be any of:
 
-* Causes a trap. That is, causes the shader invocation to write zero values to all shader stage outputs, and then terminate without other side effects.
+** 0, 1, or the maximum representable positive integer value, for
```

Comment:
nit: markdown syntax is to indent the * instead of doubling it. Same for the next bullet point